lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -224,13 +232,194 @@ def __init__(self, *args):
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 3
+LIBPATCH = 4
 
 # Alert rules directory in workload container
 RULES_DIR = "/loki/rules"
 
 logger = logging.getLogger(__name__)
 
+RELATION_INTERFACE_NAME = "loki_push_api"
+DEFAULT_RELATION_NAME = "logging"
+DEFAULT_ALERT_RULES_RELATIVE_PATH = "./src/loki_alert_rules"
+
+
+class RelationNotFoundError(Exception):
```

Comment:
Consider inheriting from ValueError or RuntimeError. Applies to the ones below as well.